autodqm/dqm.py

```diff
@@ -41,78 +54,135 @@ def __init__(self, cert, db, cache=None, workers=16):
+
+            isOnline = (series.startswith('000') and series.endswith('xxxx'))
+            if isOnline:  ## Use cmsweb.cern.ch/dqm/offline/data/browse/ROOT/OnlineData/
+                if not subsystem in OnlineMap.keys():
+                    raise error("dqm.py stream_run: {} not in OnlineMap".format(subsystem))
+                DQM_dir = OnlineMap[subsystem]
+                run_info = next(r for r in runs if r.name == run and DQM_dir+'_R000' in r.full_name)
+            else:                   ## Use cmsweb.cern.ch/dqm/offline/data/browse/ROOT/OfflineData/
+                run_info = next(r for r in runs if r.name == run)
 
             for prog in self._stream_file(
                     run_info.url, run_path, chunk_size=chunk_size):
                 yield prog
 
         size = os.path.getsize(run_path)
+        if VERBOSE: print('  * run_path = %s, size = %d' % (run_path, size))
         yield StreamProg(size, size, run_path)
 
-    def fetch_series_list(self):
-        """Return DQMRows corresponding to the series available on DQM Offline."""
-        return _resolve(self._fetch_dqm_rows(DQM_URL)).data
-
-    def fetch_sample_list(self, series):
-        """Return DQMRows corresponding to the samples available under the given
-        series."""
-        series_rows = self.fetch_series_list()
-        url = next((r.url for r in series_rows if r.name == series))
-        return _resolve(self._fetch_dqm_rows(url)).data
+    def fetch_series_list(self, dqmSource):
+        """Return DQMRows corresponding to the series available on DQM Online or Offline."""
+        if VERBOSE: print('\ndqm.py fetch_series_list(%s)' % dqmSource)
+
+        if dqmSource == "Online":
+            series_list = []
+            for series in _resolve(self._fetch_dqm_rows(DQM_URL+'OnlineData/original/')).data:
+                if series.name.startswith('000') and series.name.endswith('xxxx'):
+                    series_list.append(series)
+            return series_list
+        elif dqmSource == "Offline":
+            return _resolve(self._fetch_dqm_rows(DQM_URL+'OfflineData/')).data
+        else:
+            raise error("dqm.py fetch_series_list dqmSource = {}, not Onilne or Offline!".format(dqmSource))
 
-    def fetch_run_list(self, series, sample):
-        """Return DQMRows corresponding to the runs available under the given
-        series and sample."""
-        sample_rows = self.fetch_sample_list(series)
-        sample_url = next((r.url for r in sample_rows if r.name == sample))
+    def fetch_sample_list(self, dqmSource, series):
+        """Return DQMRows corresponding to the samples available under the given series."""
+        if VERBOSE: print('\ndqm.py fetch_sample_list(dqmSource = %s, series = %s)' % (dqmSource, series))
 
-        # Get all run directories for this sample
-        macrorun_rows = _resolve(self._fetch_dqm_rows(sample_url)).data
+        isOnline = (series.startswith('000') and series.endswith('xxxx'))
```

Comment:
This function is passed the dqmsource. Why do we need this isOnline variable?

Reply:
Indeed, previously this "isOnline" switch led to different behavior, but in subsequent development there came to be no difference in the fetch_sample_list function.  I will delete this.

Reply:
Ok, great. Once removed this section is fine.